packages/cloud/src/utils/url.ts

```diff
@@ -0,0 +1,22 @@
+import path from 'node:path';
+
+export const normalizePath = (pathLike: string) => {
+  const value = path.normalize(pathLike);
+
+  return value.length > 1 && value.endsWith('/') ? value.slice(0, -1) : value;
+};
+
+export const matchPathname = (toMatch: string, pathname: string) => {
+  const toMatchPathname = normalizePath(toMatch);
+  const normalized = normalizePath(pathname);
+
+  if (normalized === toMatchPathname) {
+    return '/';
+  }
+
+  if (normalized.startsWith(toMatchPathname + '/')) {
+    return normalized.slice(toMatchPathname.length);
+  }
+
+  return false;
```

Comment:
This makes the function return type as `string | boolean`. This might cause some issues if we use the return value in a string literal template.
How about returning an empty string instead?

Reply:
string can explicitly indicate a match, it's also good to ensure we always check the return value before use it as a string